<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>knowledge_editor</class>
 <widget class="QDialog" name="knowledge_editor">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>600</width>
    <height>500</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>Knowledge Editor</string>
  </property>
  <layout class="QVBoxLayout" name="verticalLayout">
   <item>
    <widget class="QLabel" name="title_label">
     <property name="text">
      <string>Knowledge Fact Editor</string>
     </property>
     <property name="styleSheet">
      <string>font-size: 16px; font-weight: bold;</string>
     </property>
    </widget>
   </item>
   <item>
    <layout class="QFormLayout" name="formLayout">
     <item row="0" column="0">
      <widget class="QLabel" name="text_label">
       <property name="text">
        <string>Fact Text:</string>
       </property>
      </widget>
     </item>
     <item row="0" column="1">
      <widget class="QTextEdit" name="text_input">
       <property name="maximumSize">
        <size>
         <width>16777215</width>
         <height>100</height>
        </size>
       </property>
       <property name="placeholderText">
        <string>Enter the factual information...</string>
       </property>
      </widget>
     </item>
     <item row="1" column="0">
      <widget class="QLabel" name="category_label">
       <property name="text">
        <string>Category:</string>
       </property>
      </widget>
     </item>
     <item row="1" column="1">
      <widget class="QComboBox" name="category_input">
       <property name="editable">
        <bool>true</bool>
       </property>
      </widget>
     </item>
     <item row="2" column="0">
      <widget class="QLabel" name="tags_label">
       <property name="text">
        <string>Tags:</string>
       </property>
      </widget>
     </item>
     <item row="2" column="1">
      <widget class="QLineEdit" name="tags_input">
       <property name="placeholderText">
        <string>Comma-separated tags (e.g., preference, important)</string>
       </property>
      </widget>
     </item>
     <item row="3" column="0">
      <widget class="QLabel" name="confidence_label">
       <property name="text">
        <string>Confidence:</string>
       </property>
      </widget>
     </item>
     <item row="3" column="1">
      <layout class="QHBoxLayout" name="confidence_layout">
       <item>
        <widget class="QSlider" name="confidence_slider">
         <property name="minimum">
          <number>0</number>
         </property>
         <property name="maximum">
          <number>100</number>
         </property>
         <property name="value">
          <number>80</number>
         </property>
         <property name="orientation">
          <enum>Qt::Horizontal</enum>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QLabel" name="confidence_value_label">
         <property name="text">
          <string>0.80</string>
         </property>
         <property name="minimumSize">
          <size>
           <width>50</width>
           <height>0</height>
          </size>
         </property>
        </widget>
       </item>
      </layout>
     </item>
     <item row="4" column="0">
      <widget class="QLabel" name="source_label">
       <property name="text">
        <string>Source:</string>
       </property>
      </widget>
     </item>
     <item row="4" column="1">
      <widget class="QLineEdit" name="source_input">
       <property name="placeholderText">
        <string>Where did this fact come from?</string>
       </property>
      </widget>
     </item>
     <item row="5" column="1">
      <layout class="QHBoxLayout" name="checkboxes_layout">
       <item>
        <widget class="QCheckBox" name="verified_checkbox">
         <property name="text">
          <string>Verified</string>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QCheckBox" name="enabled_checkbox">
         <property name="text">
          <string>Enabled</string>
         </property>
         <property name="checked">
          <bool>true</bool>
         </property>
        </widget>
       </item>
      </layout>
     </item>
    </layout>
   </item>
   <item>
    <widget class="QDialogButtonBox" name="buttonBox">
     <property name="standardButtons">
      <set>QDialogButtonBox::Cancel|QDialogButtonBox::Save</set>
     </property>
    </widget>
   </item>
  </layout>
 </widget>
 <resources/>
 <connections>
  <connection>
   <sender>buttonBox</sender>
   <signal>accepted()</signal>
   <receiver>knowledge_editor</receiver>
   <slot>accept()</slot>
  </connection>
  <connection>
   <sender>buttonBox</sender>
   <signal>rejected()</signal>
   <receiver>knowledge_editor</receiver>
   <slot>reject()</slot>
  </connection>
  <connection>
   <sender>confidence_slider</sender>
   <signal>valueChanged(int)</signal>
   <receiver>knowledge_editor</receiver>
   <slot>on_confidence_changed(int)</slot>
  </connection>
 </connections>
 <slots>
  <slot>on_confidence_changed(int)</slot>
 </slots>
</ui>
